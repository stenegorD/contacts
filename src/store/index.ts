import { combineReducers, configureStore } from "@reduxjs/toolkit";
import { contactsApi } from "./reducers/contacts/api";


const rootReducer = combineReducers({
    // contacts: contactsSlice.reducer,
    [contactsApi.reducerPath]: contactsApi.reducer,
});



export const store = configureStore({
    reducer: rootReducer,
    middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware().concat(
            contactsApi.middleware,
        )
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
