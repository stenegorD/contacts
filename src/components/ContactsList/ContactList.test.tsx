import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { describe, expect, it } from 'vitest';
import { ContactList } from '.';
import { store } from '../../store';


describe('ContactList', () => {
    it('should render ContactList component correctly', () => {
        const { container } = render(
            <Provider store={store}>
                <MemoryRouter>
                    <ContactList />
                </MemoryRouter>
            </Provider>
        );

        expect(container).toMatchSnapshot();
    });
});
