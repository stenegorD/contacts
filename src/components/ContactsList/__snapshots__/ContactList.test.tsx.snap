// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContactList > should render ContactList component correctly 1`] = `
<div>
  <div
    class="flex flex-col gap-2"
  >
    <h2
      class="text-xl"
    >
      Contact List
    </h2>
    <div
      class="w-full md:w-contact-list"
    >
      Loading...
    </div>
  </div>
</div>
`;
